client/lib/src/process_event.rs

```diff
@@ -19,38 +17,71 @@ pub fn process_event<UpdateServer>(
   view: &mut view::T,
   client: &client::T,
   event: Event,
+  last_key: &mut Option<VirtualKeyCode>,
 ) where UpdateServer: FnMut(protocol::ClientToServer)
 {
   match event {
-    Event::KeyDown{keycode, repeat, ..} => {
-      keycode.map(|keycode| {
-        if !repeat {
-          key_press(update_server, view, client, keycode);
+    Event::WindowEvent { event: WindowEvent::CursorMoved { .. }, .. } => {}
+    Event::WindowEvent { event: WindowEvent::MouseWheel { .. }, .. } => {}
+    Event::WindowEvent { event: WindowEvent::AxisMotion { .. }, .. } => {}
+    Event::DeviceEvent { event: DeviceEvent::Motion { .. }, .. } => {}
+    Event::DeviceEvent { event: DeviceEvent::MouseMotion { .. }, .. } => {}
+    _ => warn!("event: {:?}", event),
+  }
+
+  match event {
+    Event::WindowEvent { event, .. } => {
+      match event {
+        WindowEvent::Focused(false) => {
+          if let Some(last_key) = last_key.take() {
+            key_release(client.player_id, update_server, last_key);
+          }
         }
-      });
-    },
-    Event::KeyUp{keycode, repeat, ..} => {
-      keycode.map(|keycode| {
-        if !repeat {
-          key_release(client.player_id, update_server, keycode);
+        WindowEvent::KeyboardInput { input, .. } => {
+          if let Some(virt_keycode) = input.virtual_keycode {
+            match input.state {
+              ElementState::Pressed => {
+                if last_key == &None {
+                  key_press(update_server, view, client, virt_keycode);
+                  *last_key = Some(virt_keycode);
+                } else if last_key.is_some() && last_key != &Some(virt_keycode) {
```

Comment:
This does unexpected things when more than one key is being pressed. I think that in general it's impossible to keep track of the state of many independent keys with a single variable. It would need to be a hashtable or something similar (or we could use the SDL event system which exposes the notion of a repeated keypress without us having to reinvent it on top of glutin).

Reply:
I dont know how to use glutin and SDL together. (if possible)

Reply:
Out of curiosity, what was the issue with having just the SDL code before? I'm all for reducing complexity by using a common library, but since glutin doesn't seem to be saving any complexity here I'd argue for removing it.